mesonbuild/compilers/detect.py

```diff
@@ -572,6 +577,36 @@ def sanitize(p: str) -> str:
             return cls(
                 ccache, compiler, version, for_machine, is_cross, info,
                 exe_wrap, full_version=full_version, linker=linker)
+        
+        if 'Metrowerks C/C++' in out or 'Freescale C/C++' in out:
+            if 'ARM' in out:
+                cls = c.MetrowerksCCompilerARM if lang == 'c' else cpp.MetrowerksCPPCompilerARM
+                lnk = linkers.MetrowerksLinkerARM
+            else:
+                cls = c.MetrowerksCCompilerEmbeddedPowerPC if lang == 'c' else cpp.MetrowerksCPPCompilerEmbeddedPowerPC
+                lnk = linkers.MetrowerksLinkerEmbeddedPowerPC
+
+            mwcc_ver_match = re.search(r'Version (\d+)\.(\d+)\.?(\d+)? build (\d+)', out)
```

Comment:
```suggestion
            mwcc_ver_match = re.search(r'Version (\d+)\.(\d+)(?:\.(\d+))? build (\d+)', out)
```
(?: ... ) is a non-capturing group. Because I guess you dont want the number to finish with a `.`

Reply:
If the version doesn't have a third number, the final dot also won't be in the compiler's output, so I think the regex is ok as is?

Reply:
Yes. It is just that it would also match `1.2.`, but probably it cannot happen....